src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,49 +8,82 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "block, some street, unit number, postalcode";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final Postal postal;
     private boolean isPrivate;
 
     /**
      * Validates given address.
+     * @param postalName 
+     * @param unitName 
+     * @param streetName 
      *
      * @throws IllegalValueException if given address string is invalid.
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
+    public Address(String blockName, String streetName, String unitName, String postalName, boolean isPrivate) {
+ 
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+        this.block = new Block(blockName);
+        this.street = new Street(streetName);
+        this.unit = new Unit(unitName);
+        this.postal = new Postal(postalName);
+        this.value = block.getName()+" ,"+street.getName()+" ,"+unit.getName()+" ,"+postal.getName();
+        
     }
 
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
+   /* public Address(String value, boolean isPrivate) {
+        String[] fullAddress = value.split("'");
```

Comment:
please remove unnecessary code; you can always get back the code you wanted from a previous commit. RCS \#FTW :-D